src/realm/alloc.cpp

```diff
@@ -149,6 +149,9 @@ char* Allocator::translate_less_critical(RefTranslation* ref_translation_ptr, re
     }
     if (REALM_LIKELY(!crosses_mapping)) {
         // Array fits inside primary mapping, no new mapping needed.
+#if REALM_ENABLE_ENCRYPTION
```

Comment:
If I'm reading this correctly, I think you can flip the direction of the `if` condition, delete this block, and lift the `}` on line 173 up to before 169.

Reply:
no. The arguments to the encryption_read_barrier call is subtly different.

Reply:
I swear I stared at them several times to check that they were the same, yet somehow missed the `xover_` prefix. 🤦 

I guess you could push the `return addr;` out of the conditional. Or move the `else` block out of the `else` since the `if` ends in an early return. I'll leave it to your discretion which if any changes to make.